packages/makeup-navigation-emitter/src/index.js

```diff
@@ -1,77 +1,180 @@
 import * as KeyEmitter from 'makeup-key-emitter';
 import * as ExitEmitter from 'makeup-exit-emitter';
 
-const dataSetKey = 'data-makeup-index';
-
-// todo: autoInit: -1, autoReset: -1 are used for activeDescendant behaviour. These values can be abstracted away with
-// a new "type" option (roving or active)
-
 const defaultOptions = {
     axis: 'both',
-    autoInit: 0,
-    autoReset: null,
-    ignoreButtons: false,
+    autoInit: 'interactive',
+    autoReset: 'current',
+    ignoreByDelegateSelector: null,
     wrap: false
 };
 
-const itemFilter = (el) => !el.hidden;
+function isItemNavigable(el) {
+    return !el.hidden && el.getAttribute('aria-disabled') !== 'true';
+}
 
-function clearData(els) {
-    els.forEach(el => el.removeAttribute(dataSetKey));
+function isIndexNavigable(items, index) {
+    return (index >= 0 && index < items.length) ? isItemNavigable(items[index]) : false;
 }
 
-function setData(els) {
-    els.forEach((el, index) => el.setAttribute(dataSetKey, index));
+function findNavigableItems(items) {
+    return items.filter(isItemNavigable);
 }
 
-function isButton(el) {
-    return el.tagName.toLowerCase() === 'button' || el.type === 'button';
+function findFirstNavigableIndex(items) {
+    return items.indexOf(findNavigableItems(items)[0]);
 }
 
-function onKeyPrev(e) {
-    if (isButton(e.detail.target) === false || this.options.ignoreButtons === false) {
-        if (!this.atStart()) {
-            this.index--;
-        } else if (this.options.wrap) {
-            this.index = (this.filteredItems.length - 1);
+function findLastNavigableIndex(items) {
+    // todo: at(-1) is more performant than reverse(), but Babel is not transpiling it
+    return items.indexOf(findNavigableItems(items).reverse()[0]);
```

Comment:
```suggestion
    return items.findLastIndex((item) => isItemNavigable(item));
```

Reply:
I'll skip this one for now, `findLastIndex` fails our Tier A browser support (Safari thje culprit) and not being transpiled by Babel either (haven't looked into why): https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Array/findLastIndex